src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -384,6 +355,43 @@ def execute(self, args):
             table_transformer=self.commands_loader.command_table[parsed_args.command].table_transformer,
             is_query_active=self.data['query_active'])
 
+    def resolve_warnings(self, cmd, parsed_args):
```

Comment:
The event I'm thinking of is `EVENT_CLI_PRE_EXECUTE`.

Reply:
If we can, we *should* use the event. Ideally *all* of the core execution logic would reside in knack. The events framework is intended not just for other modules to hook into the core logic, but also to minimize the need to override the base implementation of knack.

Reply:
However, this is good for right now.

Reply:
I don't think we should be using events for this... sounds like something that would lead to problems with clear control flow. If anything, dependency injection seems like the better pattern for the core to depend on knack.

Let's talk more about this moving forward, from my POV, events should be reserved for more loosely coupled constructs that have no order dependence.

Reply:
@yugangw-msft for awareness